code/datums/elements/point_of_interest.dm

```diff
@@ -5,9 +5,21 @@
 /datum/element/point_of_interest/Attach(datum/target)
 	if (!isatom(target))
 		return ELEMENT_INCOMPATIBLE
-	GLOB.poi_list += target
+
+	// Do not ever attach to players in the lobby.
+	if(isnewplayer(target))
+		return ELEMENT_INCOMPATIBLE
```

Comment:
ELEMENT_INCOMPATIBLE *runtimes* if I remember right, you should just make sure the AddElement call never runs for new player mobs.

Reply:
And if you *do* want it to runtime (I don't think you should, I think AddElement just needs to be prevented), explain better in your comment why you are choosing to do this.

Reply:
It's a CRASH/intended runtime.

```
	if(ele.Attach(arglist(arguments)) == ELEMENT_INCOMPATIBLE)
		CRASH("Incompatible [arguments[1]] assigned to a [type]! args: [json_encode(args)]")
```

/mob/dead/new_players are abstract mobs. They are not and should not be a valid potential POI. This is a runtime error state if something attempts to turn a new player into a POI for whatever reason.

It's important it CRASHes in _AddElement because it means it fails the CI if it happens during unit tests (although there are currently no unit tests that test for that behaviour).

It also leaves a message in the runtime logs instead of failing silently - `if(this_shouldnt_happen) return` is a pet peeve of mine. You can see the full trace of the call stack that attempted to add the element to the new_player in the runtime logs, and fix it.

I'll expand the comment to explain my reasoning more.

Also, there are already additional states catching this elsewhere in the code. In addition to not being able to turn new_players into POIs here, the new SS prevents them being incuded as part of the mob POI list with a stack_trace and do_observe also sends a stack_trace if you attempt to observe a new_player. There is redundancy here, but I feel it's important this never regresses.

Reply:
As long as you are already preventing new players from getting POIs, then this is fine.